### PR TITLE
[ui] Themes

### PR DESCRIPTION
**Pull request by hellendag**
## Summary & Motivation

Introduce a system for theming the Dagster UI. All direct references to color values are replaced by a set of functions that match semantic color names to theme-based palette values.

The theme value is exposed via a User Setting.

Discussion here: https://github.com/dagster-io/dagster/discussions/18439

## How I Tested These Changes

TS, lint, jest. Find all existing references to `LegacyColors`, hex values, rgb/rgba values and replace them with color functions.

Navigate throughout the app in legacy, light, and dark modes. Verify that colors look appropriate.

### CODE REVIEW

**Review by bengotow — APPROVED** [review]
This looks awesome. The approach seems solid and I'm a big fan of the semantic naming of the color values.  Props for going through and re-classifying everything, this must have taken a while! 💯

File: js_modules/dagster-ui/packages/ui-core/src/asset-graph/AssetNode.tsx [hunk 1]
```diff
@@ -400,19 +422,21 @@ const AssetNodeShowOnHover = styled.span`
...
-  border-radius: 8px;
+  background: ${colorBackgroundDefault()};
+  border-radius: 10px;
   position: relative;
+  transition: all 150ms linear;
   &:hover {
+    ${(p) => !p.$selected && `border: 2px solid ${colorLineageNodeBorderHover()};`};
+
     box-shadow: rgba(0, 0, 0, 0.12) 0px 2px 12px 0px;
+    scale: 1.03;
```
[the remark this answers is not in the document]
Comment (braunjj) [reply js_modules/dagster-ui/packages/ui-core/src/asset-graph/AssetNode.tsx:439]: I did added this one. It adds a slight scale effect on hover. We have the same effect on the new group nodes as well. happy to revisit this if y'all find it distracting but the primary motivation for adding it was that we previously relied on a box shadow for the node hover style, and this doesn't show up well in dark mode. The scale transition works well in both light and dark mode

File: js_modules/dagster-ui/packages/app-oss/src/NUX/CommunityNux.tsx [hunk 2]
```diff
@@ -101,7 +103,7 @@ const Form = ({dismiss, submit}: FormProps) => {
       >
         <Box flex={{direction: 'column', gap: 8, alignItems: 'start', justifyContent: 'start'}}>
           <Heading>Join the Dagster community</Heading>
-          <Body style={{color: Colors.Gray700, marginBottom: '4px'}}>
```
[the remark this answers is not in the document]
Comment (hellendag) [reply js_modules/dagster-ui/packages/app-oss/src/NUX/CommunityNux.tsx:104]: Yeah, I guess we could have the functions namespaced under `Colors`. If you guys feel strongly about doing that instead, we could do a find/replace for it.

File: js_modules/dagster-ui/packages/ui-components/src/components/Checkbox.tsx [hunk 3]
```diff
@@ -35,31 +45,34 @@ const StarIcon = ({checked, indeterminate, fillColor, disabled}: IconProps) => (
     <path
       className="interaction-focus-outline"
       d="M8.99983 14.27L13.1498 16.78C13.9098 17.24 14.8398 16.56 14.6398 15.7L13.5398 10.98L17.2098 7.80001C17.8798 7.22001 17.5198 6.12001 16.6398 6.05001L11.8098 5.64001L9.91983 1.18001C9.57983 0.37001 8.41983 0.37001 8.07983 1.18001L6.18983 5.63001L1.35983 6.04001C0.479829 6.11001 0.119828 7.21001 0.789828 7.79001L4.45983 10.97L3.35983 15.69C3.15983 16.55 4.08983 17.23 4.84983 16.77L8.99983 14.27Z"
-      fill={Colors.White}
+      fill={disabled ? colorBackgroundGray() : colorBackgroundDefault()}
```
[the remark this answers is not in the document]
Comment (hellendag) [reply js_modules/dagster-ui/packages/ui-components/src/components/Checkbox.tsx:48]: I don't think I've seen the 5pm shift before! I'm just always in dark mode. We could come back to this if it seems important to users.

**Pull request merged**